### ISSUE TITLE
Arty marker distance bugs when replacing or renaming (moving works fine)

### ISSUE THREAD

**Issue by RemoteControlledArty** [RemoteControlledArty/RemoteControlledArtillery#17]
Distance bugs to 29998m when replacing or renaming it (while moving it works fine).
Cycling markers with 5 fixes distance, but unnoticed leads to misfires.

Example: arty1 marker is placed, engaged and removed, even minutes later if arty1 is placed again, the distance is bugged.

**Comment by Tapawingo** [comment]
Would that also incidentally be the distance to [0, 0, 0]? If it is, then this is probably an indexing issue. I see that in `RC_UI_Loop` you are passing the markers into a GVAR called `RC_Current_Target` and then this GVAR is used later when calculating the distance. Most likely, when you are renaming or replacing the marker the one stored in `RC_Current_Target` is not modified. So when you are trying to get the marker position it returns [0, 0, 0] as the stored marker no longer exists.

**Comment by realfluffyuwu** [comment]
After having a quick read through again I believe I accounted for situations where the marker didn't exist anymore but maybe I just didn't test it well enough?

[This](https://github.com/RemoteControlledArty/RemoteControlledArtillery/blob/653fd8c5a92290c0655fb7aaabfa95bc4bd67082/addons/Remote_Controlled_Artillery/functions/fn_RC_UI_Loop.sqf#L264) is checking for `isNil` which should reset it to index 0 but perhaps changing this to checking if the markers Position is at [0, 0, 0] which is always returned if the marker doesn't exist might prove better for this case.

Although this still doesn't solve needing to cycle through the markers to acquire the correct target.

**Comment by Tapawingo** [comment]
> After having a quick read through again I believe I accounted for situations where the marker didn't exist anymore but maybe I just didn't test it well enough?
> 
> [This](https://github.com/RemoteControlledArty/RemoteControlledArtillery/blob/653fd8c5a92290c0655fb7aaabfa95bc4bd67082/addons/Remote_Controlled_Artillery/functions/fn_RC_UI_Loop.sqf#L264) is checking for `isNil` which should reset it to index 0 but perhaps changing this to checking if the markers Position is at [0, 0, 0] which is always returned if the marker doesn't exist might prove better for this case.
> 
> Although this still doesn't solve needing to cycle through the markers to acquire the correct target.

Yeah this should account for it. I think if you can replicate the bug you should check the value of `RC_Current_Target`. This should give us a better idea of what is happening under the hood.

*Edit: I just tested it and `RC_Current_Target` is not updated to the new marker 😬*
